KioskApplication/src/main/java/controller/MainWindowController.java

```diff
@@ -178,32 +189,56 @@ public void changed(ObservableValue<? extends Tab> ov, Tab oldValue, Tab newValu
 //        });
     }
 
+    protected void removeLoginPopup(){
+        this.loginPopup.getChildren().clear();
+        this.switchButton.setDisable(false);
+        this.tabMap.setDisable(false);
+        this.mapView.setDisable(false);
+        //this.loginPopup.getChildren().add(this.switchButton);
+    }
+
     @FXML
-    private void login() throws IOException{
+    private void addLoginPopup() throws IOException{
+        this.switchButton.setDisable(true);
+        this.tabMap.setDisable(true);
+        this.mapView.setDisable(true);
         LoginController loginController = new LoginController(this);
-        FXMLLoader loader = new FXMLLoader(getClass().getResource("/view/AdminLoginWindow.fxml"));
+        FXMLLoader loader = new FXMLLoader(getClass().getResource("/view/LoginWindow.fxml"));
         loader.setController(loginController);
         javafx.scene.Node view = loader.load();
 
-        BorderPane loginContainer = new BorderPane();
-        AnchorPane.setTopAnchor(loginContainer, 0.0);
-        AnchorPane.setLeftAnchor(loginContainer, 0.0);
-        AnchorPane.setBottomAnchor(loginContainer, 0.0);
-        AnchorPane.setRightAnchor(loginContainer, 0.0);
-
-        loginContainer.setCenter(view);
-
-        contentWindow.getChildren().add(loginContainer);
+        this.loginPopup = new BorderPane();
+        AnchorPane.setTopAnchor(loginPopup, 0.0);
+        AnchorPane.setLeftAnchor(loginPopup, 0.0);
+        AnchorPane.setBottomAnchor(loginPopup, 0.0);
+        AnchorPane.setRightAnchor(loginPopup, 0.0);
+
+        loginPopup.setRight(view);
+        contentWindow.getChildren().add(loginPopup);
+
```

Comment:
This is really good, but you should only load the FXML for the login file once. No reason to do a file read every time the user clicks the button. Maybe do that in initialize and then do the adding logic here.